Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.App.KronosWfc.BusinessLogic/TimeOff/TimeOffActivity.cs

```diff
@@ -141,7 +175,84 @@ public TimeOffActivity(TelemetryClient telemetryClient, IApiHelper apiHelper)
                     { "Response", tupleResponse.Item1 },
                 });
 
-            return tupleResponse.ProcessResponse<ApproveDeclineResponse>(this.telemetryClient);
+            return tupleResponse.ProcessResponse<CommonResponse>(this.telemetryClient);
+        }
+
+        /// <inheritdoc/>
+        public async Task<CommonResponse> AddManagerCommentsToTimeOffRequestAsync(
+            Uri endPointUrl,
+            string jSession,
+            string kronosRequestId,
+            DateTimeOffset startDateTime,
+            DateTimeOffset endDateTime,
+            string queryDateSpan,
+            string personNumber,
+            string reason,
+            string managerMessage,
+            string managerCommentText,
+            Comments existingNotes)
+        {
+            var comments = this.AddTimeOffRequestNotes(managerMessage, managerCommentText, existingNotes.Comment);
+            var xmlTimeOffRequestUpdateRequest = this.CreateUpdateTimeOffRequest(kronosRequestId, startDateTime, endDateTime, queryDateSpan, personNumber, reason, comments);
+
+            var tupleResponse = await this.apiHelper.SendSoapPostRequestAsync(
+                endPointUrl,
+                ApiConstants.SoapEnvOpen,
+                xmlTimeOffRequestUpdateRequest,
+                ApiConstants.SoapEnvClose,
+                jSession).ConfigureAwait(false);
+
+            this.telemetryClient.TrackTrace(
+                "TimeOffActivity - ApproveOrDenyTimeOffRequestAsync",
+                new Dictionary<string, string>()
+                {
+                    { "Response", tupleResponse.Item1 },
+                });
+
+            return tupleResponse.ProcessResponse<CommonResponse>(this.telemetryClient);
+        }
+
+        /// <summary>
+        /// This method will calculate the necessary duration of the time off.
+        /// </summary>
+        /// <param name="startDateTime">The start date/time stamp for the time off request.</param>
+        /// <param name="endDateTime">The end date/time stamp for the time off request.</param>
+        /// <param name="reason">The time off reason from Shifts.</param>
+        /// <returns>A string that represents the duration period.</returns>
+        private static List<TimeOffPeriod> CalculateTimeOffPeriod(DateTimeOffset startDateTime, DateTimeOffset endDateTime, string reason)
```

Comment:
The Kronos API expects a list of time off periods but you are right, this is misleading. I will update.

Reply:
I think I understand, is it because the request object submitted to Kronos is expecting a collection.

Reply:
It does, but I think it would make more sense to just return a single time off period and construct the collection in the calling method to save on confusion.